File: py/examples/stat_large_bar.py
# Stat / Bar / Large
# Create a large captioned card displaying a primary value, an auxiliary value and a progress bar,
# with captions for each value.
# #stat_card #progress
# ---
import time
from h2o_wave import site, ui

price_changes = [(62.54, 0.2345), (9.42, 0.4033), (26.73, 0.3065), (31.44, 0.10000), (52.90, 0.482),
                 (8.25, 0.2650), (17.12, 0.0328), (20.35, 0.4638), (48.76, 0.3756), (30.14, 0.1500), ]


def next_price_fluctuation():
    i = 0
    while True:
        yield price_changes[i % len(price_changes)]
        i += 1


page = site['/demo']
price_change = next_price_fluctuation()
price, pc = next(price_change)

stat_card = page.add(f'example', ui.large_bar_stat_card(
    box='1 1 2 2',
    title='Donut price fluctuations',
    value='=${{intl price minimum_fraction_digits=2 maximum_fraction_digits=2}}',
    value_caption='This Month',
    aux_value='={{intl percentage style="percent" minimum_fraction_digits=2 maximum_fraction_digits=2}}',
    aux_value_caption='Previous Month',
    plot_color='$red',
    progress=pc,
    data=dict(price=price, percentage=pc),
    caption='Fluctuations of the donuts price over the time',
))
page.save()

while True:
    time.sleep(1)
    price, pc = next(price_change)
    stat_card.data.price = price
    stat_card.data.percentage = pc
    stat_card.progress = pc
    page.save()

File: r/R/examples/stat_large_bar.R
library(h2owave)

dark_theme_colors = c('$red', '$pink', '$blue', '$azure', '$cyan', '$teal', '$mint', '$green', '$lime', '$yellow', '$amber', '$orange' ,'$tangerine')
curves = c('linear','smooth','step','step-after','step-before')
crypto_name <- c("ETH","BTC","ZCH","LTC")

sample_crypto_name <- crypto_name[sample(1:length(crypto_name),1)]
sample_crypto_price <- runif(1,1,500)
sample_crypto_price_percent_change <- runif(1,0,1)
card_color <- sample(dark_theme_colors,1)
caption <- paste0("The card shows the price (in USD) and price percentage change of ",sample_crypto_name,".")

page <- Site("/demo")
page$drop()
page$add_card("large_bar",ui_large_bar_stat_card(box="1 1 2 2"
                                                                          ,title=sample_crypto_name
                                                                          ,value='=${{intl price minimum_fraction_digits=2 maximum_fraction_digits=2}}'
                                                                          ,value_caption='This Month'
                                                                          ,aux_value='={{intl change style="percent" minimum_fraction_digits=2 maximum_fraction_digits=2}}'
                                                                          ,aux_value_caption='Previous Month'
                                                                          ,plot_color=card_color
                                                                          ,data=list(price=sample_crypto_price,change=sample_crypto_price_percent_change)
                                                                          ,progress = sample_crypto_price_percent_change
                                                                          ,caption=caption
))

page$save()

while(TRUE){
  Sys.sleep(3)
  sample_crypto_price <- runif(1,1,500)
  sample_crypto_price_percent_change <- runif(1,0,1)
  page$set("large_bar","data","price",sample_crypto_price)
  page$set("large_bar","data","change",sample_crypto_price_percent_change)
  page$set("large_bar","progress",sample_crypto_price_percent_change)
  page$save()
}
